packages/api/endpoints/rules.js

```diff
@@ -111,17 +120,28 @@ async function put({ params: { name }, body }, res) {
 
   try {
     const oldRule = await model.get({ name });
+    const dbClient = await getKnexClient();
 
-    // if rule type is onetime no change is allowed unless it is a rerun
+    // If rule type is onetime no change is allowed unless it is a rerun
     if (body.action === 'rerun') {
       return models.Rule.invoke(oldRule).then(() => res.send(oldRule));
     }
 
+    let updatedRuleItem = cloneDeep(oldRule);
     // Remove all fields from the existing rule that are not supplied in body
     // since body is expected to be a replacement rule, not a partial rule
     const fieldsToDelete = Object.keys(oldRule).filter((key) => !(key in body));
-    const newRule = await model.update(oldRule, body, fieldsToDelete);
-
+    updatedRuleItem = omit(updatedRuleItem, fieldsToDelete);
+    const record = merge(updatedRuleItem, body);
```

Comment:
I'm curious whether this is necessary for Postgres. If you have an existing rule record in Postgres, then you submit an update with fields missing, then I think the upsert query would set those missing fields to null. 

This comes down to the difference between Dynamo and Postgres. Dynamo is a schema-less database where we just store objects, so if we want a property removed, we have to do so manually as part of saving. However, in Postgres, all properties have columns that are either null or have a value. So submitting an update with a field missing to `translateApiRuleToPostgresRule` should mean that field just gets set to null, which is all we need.

Thus, I think we just need

```
 const newPostgresRecord = await translateApiRuleToPostgresRule(body, dbClient);
```

I'm curious if the test for removing a field still passes with just this logic

Reply:
Hmm good point. Looks like you are correct [`c4ffbec` (#2012)](https://github.com/nasa/cumulus/pull/2012/commits/c4ffbec1dd0be748fc8cf8163bf7e91d8a2bd357)